test/uk/gov/hmrc/minorentityidentificationfrontend/services/RegistrationOrchestrationServiceSpec.scala

```diff
@@ -81,4 +81,61 @@ class RegistrationOrchestrationServiceSpec extends AnyWordSpec with Matchers wit
     }
   }
 
+  "register Unincorporated association" should {
+    "return Registered" when {
+      "the user has successfully passed BV check" in {
+        mockStorageService.retrieveBusinessVerificationStatus(testJourneyId) returns Future.successful(Some(BusinessVerificationPass))
+        mockRegistrationConnector.registerUA(testCtutr, testRegime) returns Future.successful(Registered(testSafeId))
+        mockStorageService.storeRegistrationStatus(testJourneyId, Registered(testSafeId)) returns Future.successful(SuccessfullyStored)
+
+        mockAuditService.auditJourney(testJourneyId, testUAJourneyConfig()) returns Future.successful(())
+
+        val result = await(TestRegistrationOrchestrationService.register(testJourneyId, Some(testCtutr), testUAJourneyConfig()))
+
+        result mustBe Registered(testSafeId)
+
+        mockAuditService.auditJourney(testJourneyId, testUAJourneyConfig()) was called
+      }
+      "the Business Verification Check is disabled" in {
+        mockStorageService.retrieveBusinessVerificationStatus(testJourneyId) returns Future.successful(None)
+        mockRegistrationConnector.registerUA(testCtutr, testRegime) returns Future.successful(Registered(testSafeId))
+        mockStorageService.storeRegistrationStatus(testJourneyId, Registered(testSafeId)) returns Future.successful(SuccessfullyStored)
+
+        mockAuditService.auditJourney(journeyId = testJourneyId, journeyConfig = testUAJourneyConfig(false)) returns Future.successful(())
+
+        val result = await(TestRegistrationOrchestrationService.register(testJourneyId, Some(testCtutr), testUAJourneyConfig(false)))
+
+        result mustBe Registered(testSafeId)
+
+        mockAuditService.auditJourney(journeyId = testJourneyId, journeyConfig = testUAJourneyConfig(false)) was called
+      }
+    }
+    "return RegistrationNotCalled" when {
+      "the user did not pass BV checks" in {
+        mockStorageService.retrieveBusinessVerificationStatus(testJourneyId) returns Future.successful(Some(BusinessVerificationFail))
+        mockRegistrationConnector.registerUA(testCtutr, testRegime) returns Future.successful(RegistrationNotCalled)
+        mockStorageService.storeRegistrationStatus(testJourneyId,RegistrationNotCalled) returns Future.successful(SuccessfullyStored)
+
+        mockAuditService.auditJourney(testJourneyId, testUAJourneyConfig()) returns Future.successful(())
+
+        val result = await(TestRegistrationOrchestrationService.register(testJourneyId, Some(testCtutr), testUAJourneyConfig()))
+
+        result mustBe RegistrationNotCalled
+
+        mockAuditService.auditJourney(testJourneyId, testUAJourneyConfig()) was called
+      }
+    }
+  }
+
+  "register overseas company" should {
+    "return an exception for registration" when {
```

Comment:
throw an exception ;-) ...because exception can be also returned but this is not that case ;-)